Backpack/src/main/java/net/skyscanner/backpack/dialog/BpkDialog.kt

```diff
@@ -0,0 +1,120 @@
+package net.skyscanner.backpack.dialog
+
+import android.app.Dialog
+import android.content.Context
+import android.graphics.Color
+import net.skyscanner.backpack.R
+import net.skyscanner.backpack.text.BpkText
+import android.graphics.drawable.ColorDrawable
+import android.graphics.drawable.GradientDrawable
+import android.util.DisplayMetrics
+import android.view.ViewGroup
+import android.view.Window
+import android.view.WindowManager
+import android.view.Gravity
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.LinearLayout
+import android.widget.RelativeLayout
+import androidx.annotation.ColorRes
+import androidx.annotation.DrawableRes
+import androidx.core.content.res.ResourcesCompat
+import net.skyscanner.backpack.button.BpkButton
+
+open class BpkDialog(
+  context: Context,
+  val style: Style = Style.ALERT
+) : Dialog(context, 0) {
+
+  data class Icon(@DrawableRes val drawableRes: Int, @ColorRes val colorRes: Int)
+
+  enum class Style {
+    ALERT, BOTTOM_SHEET
+  }
+
+  private var viewHolder: ViewHolder? = null
+  private val variables = DialogVariables(context)
+
+  init {
+    setupDialog()
+  }
+
+  var title = ""
+    set(value) {
+      field = value
+      viewHolder?.title?.text = value
+    }
+
+  var description = ""
+    set(value) {
+      field = value
+      viewHolder?.description?.text = value
+    }
+
+  var icon: Icon? = null
+    set(value) {
+      field = value
+      viewHolder?.iconView?.icon = icon
+    }
+
+  fun addActionButton(button: BpkButton) {
+    button.layoutParams = LinearLayout.LayoutParams(
+      LinearLayout.LayoutParams.MATCH_PARENT,
+      LinearLayout.LayoutParams.WRAP_CONTENT
+    ).apply {
+      setMargins(0, 0, 0, variables.buttonStackItemSpace)
+    }
+    viewHolder?.contentLayout?.addView(button)
+  }
+
+  private fun setupDialog() {
+    if (style == Style.BOTTOM_SHEET) {
+      window?.setGravity(Gravity.BOTTOM)
+      window?.setWindowAnimations(R.style.Bpk_dialog_animation)
+    }
+
+    requestWindowFeature(Window.FEATURE_NO_TITLE)
+    setContentView(setUpContentView())
+    window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
+
+    val metrics = DisplayMetrics()
+    val windowManager = context.getSystemService(Context.WINDOW_SERVICE) as WindowManager
+    windowManager.defaultDisplay.getMetrics(metrics)
+    window?.setLayout((metrics.widthPixels * 0.85).toInt(), RelativeLayout.LayoutParams.WRAP_CONTENT)
```

Comment:
what is `.85` ?

Reply:
85% of the screen size. Again this is based on the IOS version.

Reply:
Again following what is done in IOS. I can add an upper limit though.

Reply:
I just removed this so it will use the default dialog size now.